src/base/rss/rssfeed.cpp

```diff
@@ -114,8 +114,21 @@ void Feed::saveItemsToDisk()
 
 void Feed::loadItemsFromDisk()
 {
-    QIniSettings qBTRSS("qBittorrent", "qBittorrent-rss");
-    QHash<QString, QVariant> allOldItems = qBTRSS.value("old_items", QHash<QString, QVariant>()).toHash();
+    QIniSettings qBTRSS("qBittorrent", "qBittorrent-rss-items");
+
+    // Copy old items to new config file if needed
+    QIniSettings qBTRSSLegacy("qBittorrent", "qBittorrent-rss");
+    QHash<QString, QVariant> allOldItems = qBTRSSLegacy.value("old_items", QHash<QString, QVariant>()).toHash();
+
+    if (!allOldItems.empty()) {
+        qDebug("Moving %d old items for feeds to new file", allOldItems.size());
+        qBTRSS.setValue("old_items", allOldItems);
+        qBTRSSLegacy.remove("old_items");
+    }
+    else {
+        allOldItems = qBTRSS.value("old_items", QHash<QString, QVariant>()).toHash();
+    }
```

Comment:
move the migrating code under a function called "migrateRSS()" in the file `src/app/upgrade.h".
Then call it inside main() at the appropriate position.
Wrap all that code in:
```c++
#ifndef DISABLE_GUI
// do stuff
#endif

After all this the code in this file will be much cleaner.
And in the future, we have to look at one place and remove old upgrading/migrating code.